### PR TITLE
Fix wp core check update in wp cli

### LINKED ISSUE #470: WP-CLI issue with `wp core check-update`
https://github.com/xxsimoxx/cpcompatibility/blob/1156ef6d98a86e902a218514bfcc4f196d47a666/cpcompatibility.php#L37-L56

Thoughts on incorporating something like this into core? @xxsimoxx @johnalarcon

Edit: we'd probably want to parse the update response more thoroughly, including showing what new version of ClassicPress is available, and also making the update actually work using WP-CLI would be good.

### PR DESCRIPTION
**Pull request by xxsimoxx**
## Description
Added a class that stops the command from being executed and implement the command to give the expected results.

## Motivation and context
`wp core check-update` is not working properly (it always shows that CP is up-to-date).
Fixes #470.

## How has this been tested?
Tested in local envirovment.
The code is from [cpcompatibility](https://github.com/xxsimoxx/cpcompatibility) which I used for more than an year on 10+ production sites.

Test with phpcs return errors for missing doc comment.

### Before
image

### After
image

### CODE REVIEW

**Comment by viktorix** [comment]
@xxsimoxx is this still needed?

**Comment by xxsimoxx** [comment]
@viktorix yes, `wp core check-update` don't work in CP 1.4.2.

**Comment by viktorix** [comment]
@bahiirwa @mattyrob is this something we can get into 1.5.0 so we don't have issues with WP-CLI?

**Comment by mattyrob** [comment]
Personally I don't use `WP-Cli` so this is not my area.

One issue I have with the solution is it is always there, all the time.

Would it still work is the class was conditionally loaded in `wp-settings.php`?

**Comment by xxsimoxx** [comment]
Agree @mattyrob. I've pushed a change to conditionally load the class.

**Comment by viktorix** [comment]
I see you already moved the check. Ensuring we keep CP compatible with WP-CLI for as long as possible is important because most hosts have WP-CLI installed and many developers do use it. Automations are set up to use WP-CLI as well on the backend.

**Comment by bahiirwa** [comment]
This works as "advertised".

In the screenshot, I have the before and after.

Screenshot 2022-07-19 at 21 02 33

**Comment by bahiirwa** [comment]
@xxsimoxx Which command are you suing to get the package URL table? I am getting `4.9.20` as latest version.
Both on local and live production server.

**Comment by xxsimoxx** [comment]
@bahiirwa the command is `wp core check-update`.
Do you mean you get `4.9.20` with `wp core version`?
To manually check the PR you can change `$cp_version` in `wp-includes/version.php`.

**Comment by bahiirwa** [comment]
Manual tests done. Works as planned.

See screenshots.

`wp core version` returns wp version because that is the right WP version defined in the `wp-includes/version.php` file

However, if `$cp_version` is defined to be of a lower version than the current release, `wp core check-update` will return the latest version and allows for the core to be updated with `wp core update` to `1.4.2` in this case. The `$cp_version` will also be updated to `1.4.2`

Screenshot 2022-07-20 at 10 05 30

Screenshot 2022-07-20 at 10 05 10

File: src/wp-includes/class-fix-wpcli.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<?php
+class Fix_WPCLI {
```
Comment (bahiirwa) [inline src/wp-includes/class-fix-wpcli.php:2]: Add class descriptions.

File: src/wp-includes/class-fix-wpcli.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<?php
```
Comment (bahiirwa) [inline src/wp-includes/class-fix-wpcli.php:1]: WPCS: Add file description

File: src/wp-includes/class-fix-wpcli.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+class Fix_WPCLI {
+
+	public function __construct() {
+		WP_CLI::add_hook( 'before_invoke:core check-update', array( __CLASS__, 'correct_core_check_update' ) );
+	}
+
+	public static function correct_core_check_update() {
```
Comment (bahiirwa) [inline src/wp-includes/class-fix-wpcli.php:8]: WPCS: Methods require descriptions

**Comment by bahiirwa** [comment]
@xxsimoxx 

We also need to fix these issues 

```
FILE: /Applications/MAMP/htdocs/ClassicPress/src/wp-includes/class-fix-wpcli.php
--------------------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 4 LINES
--------------------------------------------------------------------------------
 1 | ERROR | Missing file doc comment
 2 | ERROR | Missing doc comment for class Fix_WPCLI
 4 | ERROR | Missing doc comment for function __construct()
 8 | ERROR | Missing doc comment for function correct_core_check_update()
--------------------------------------------------------------------------------
```

File: src/wp-includes/class-fix-wpcli.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
+<?php
+class Fix_WPCLI {
+
+	public function __construct() {
```
Comment (bahiirwa) [inline src/wp-includes/class-fix-wpcli.php:4]: WPCS: Missing doc comment for function __construct()

**Comment by xxsimoxx** [comment]
Please check if DocBlock is correct, I've never used it.

**Comment by viktorix** [comment]
The fix works as intended. Here's a before/after:

image

**Comment by viktorix** [comment]
@xxsimoxx can you update wp-settings.php to resolve conflict introduced by the layout PR? It's just formatting of require().
https://github.com/ClassicPress/ClassicPress/pull/811/conflicts

**Review by viktorix — COMMENTED** [review]
This looks good. No conflicts.

**Comment by viktorix** [comment]
@mattyrob this PR should be ready for merging.

**Comment by mattyrob** [comment]
Has 2 reviews, looks like a useful change for `wp-cli` users of ClassicPress.

**Pull request merged**